contract/src/airdrop/prepare.js

```diff
@@ -0,0 +1,267 @@
+import { M } from '@endo/patterns';
+import { makeDurableZone } from '@agoric/zone/durable.js';
+import { E } from '@endo/far';
+import { Far } from '@endo/marshal';
+import { AmountMath } from '@agoric/ertp';
+import { TimeMath, TimestampShape } from '@agoric/time';
+import '../../types.js'
+
+/**
+ * @typedef {object} EpochDetails
+ * @property {bigint} windowLength Length of epoch in seconds. This value is used by the contract's timerService to schedule a wake up that will fire once all of the seconds in an epoch have elapsed
+ * @property {bigint} earlyClaimBonus The total number of tokens recieved by each user who claims during a particular epoch.
+ * @property {bigint} index The index of a particular epoch.
+ * @property {number} inDays Length of epoch formatted in total number of days
+ */
+
+const startupAssertion = (arg, keyName) =>
+  assert(
+    arg,
+    `Contract has been started without required property: ${keyName}.`,
+  );
+
+const makeWaker = (name, func) => {
+  return Far(name, {
+    wake: timestamp => func(timestamp),
+  });
+};
+
+// const createWakeup = async (timer, wakeUpTime, timeWaker, cancelTokenMaker) => {
+//   const cancelToken = cancelTokenMaker();
+//   await E(timer).setWakeup(wakeUpTime, timeWaker, cancelToken);
+// };
+const makeCancelTokenMaker = name => {
+  let tokenCount = 1;
+
+  return () => Far(`cancelToken-${name}-${(tokenCount += 1)}`, {});
+};
+
+/**
+ *
+ * @typedef {object} ContractTerms
+ * @property {import('@endo/marshal').RemotableObject} AirdropUtils
+ * @property {bigint} startTime
+ * @property {bigint} basePayoutQuantity
+ * @property {{Keyword<Brand>}} brands
+ * @property {{Keyword<issuer>}} issuers
+ */
+
+/**
+ * @param {ZCF} zcf
+ * @param {{ purse: Purse, timer: import('@agoric/swingset-vat/tools/manual-timer.js').TimerService,}} privateArgs
+ * @param {import('@agoric/vat-data').Baggage} baggage
+ */
+export const start = async (zcf, privateArgs, baggage) => {
+  const zone = makeDurableZone(baggage, 'rootZone');
+
+  const terms = zcf.getTerms();
+
+  /** @type {ContractTerms} */
+  const {
+    AirdropUtils,
+    startTime,
+    basePayoutQuantity,
+    brands: { Token: tokenBrand },
+    issuers: { Token: tokenIssuer },
+  } = terms;
+
+  const createAmount = x => AmountMath.make(tokenBrand, x);
+
+  assert(startTime > 0n, 'startTime must be a BigInt larger than 0n.');
```

Comment:
consider letting Zoe take care of that too:

```js
export const CustomTermsShape = {
  startTime: M.gte(0n),
  // etc.
};
```

Reply:
@dckc here is the IOU I promises in response to your feedback re `customtermsShaoe`.

I'm still working to get a firm grasp on using matchers (at which point I plan to implement stronger constraints.


Below is the code for `customTermsShape` and `privateArgsShape` as found within the repo's [dev branch](
https://github.com/lari-labs/dapp-ertp-airdrop/blob/df9ec04feb248e13bf06900c9448cc221ea326d4/contract/src/airdrop/prepare.js#L19-L38). 


```javascript
const privateArgsShape = harden({
  purse: PurseShape,
  timer: TimerShape,
});


const customTermsShape = harden({
  startTime: M.gte(0n),
  initialState: M.string(),
  stateTransitions: M.arrayOf(M.array()),
  states: M.record(),
  schedule: M.array(),
  basePayoutQuantity: AmountShape,
});


/** @type {import('../../types.js').ContractMeta} */
export const meta = {
  customTermsShape,
  privateArgsShape,
  upgradability: 'canUpgrade',
};
```

Am I right to believe that using this approach eliminates the need to check these values using `assert` within the body of the contract?